internal/webhook/v1/aerospikecluster_validating_webhook.go

```diff
@@ -714,206 +706,41 @@ func validateClusterSize(_ logr.Logger, sz int) error {
 }
 
 func validateAerospikeConfig(
-	configSpec *asdbv1.AerospikeConfigSpec, storage *asdbv1.AerospikeStorageSpec, clSize int,
+	aslog logr.Logger, version string, configSpec *asdbv1.AerospikeConfigSpec,
+	storage *asdbv1.AerospikeStorageSpec, clSize int,
 	clientCert *asdbv1.AerospikeOperatorClientCertSpec,
 ) error {
-	config := configSpec.Value
-
-	if config == nil {
-		return fmt.Errorf("aerospikeConfig cannot be empty")
-	}
-
-	// service conf
-	serviceConf, ok := config["service"].(map[string]interface{})
-	if !ok {
-		return fmt.Errorf(
-			"aerospikeConfig.service not a valid map %v", config["service"],
-		)
-	}
-
-	if val, exists := serviceConf["advertise-ipv6"]; exists && val.(bool) {
-		return fmt.Errorf("advertise-ipv6 is not supported")
-	}
-
-	if _, ok = serviceConf["cluster-name"]; !ok {
-		return fmt.Errorf("aerospikeCluster name not found in config. Looks like object is not mutated by webhook")
-	}
-
-	// network conf
-	networkConf, ok := config["network"].(map[string]interface{})
-	if !ok {
-		return fmt.Errorf(
-			"aerospikeConfig.network not a valid map %v", config["network"],
-		)
-	}
-
-	if err := validateNetworkConfig(networkConf, clientCert); err != nil {
-		return err
-	}
-
-	// namespace conf
-	nsListInterface, ok := config["namespaces"]
-	if !ok {
-		return fmt.Errorf(
-			"aerospikeConfig.namespace not a present. aerospikeConfig %v",
-			config,
-		)
-	} else if nsListInterface == nil {
-		return fmt.Errorf("aerospikeConfig.namespace cannot be nil")
-	}
-
-	if nsList, ok1 := nsListInterface.([]interface{}); !ok1 {
-		return fmt.Errorf(
-			"aerospikeConfig.namespace not valid namespace list %v",
-			nsListInterface,
-		)
-	} else if err := validateNamespaceConfig(
-		nsList, storage, clSize,
+	// It validates the aerospikeConfig schema and generic aerospikeConfig fields
+	if err := validation.ValidateAerospikeConfig(
```

Comment:
When we started passing a value instead of a configSpec pointer, checking for nil on the map might incorrectly allow an empty map. Additionally, if we type-cast a value from the map without first verifying that the corresponding key exists, this could potentially lead to a runtime failure. Have we considered whether any such scenarios could occur?